packages/dialect-react-sdk/hooks/useSingleFeed.ts

```diff
@@ -0,0 +1,51 @@
+import type { DappMessage, FindDappMessageQuery } from '@dialectlabs/sdk';
+import { useMemo } from 'react';
+import useSWR from 'swr';
+import { EMPTY_ARR } from '../utils';
+import { SINGLE_FEED_CACHE_KEY_FN } from './internal/swrCache';
+import useDialectSdk from './useDialectSdk';
+
+interface UseSingleFeedParams {
+  nonVerifiedDapps?: boolean;
+  take?: number;
+  skip?: number;
+  pollingInterval?: number;
+}
+
+interface UseSingleFeedValue {
+  data: DappMessage[];
+  isLoading: boolean;
+}
+
+const useSingleFeed = ({
+  nonVerifiedDapps = false,
+  take = 25,
+  skip = 0,
+  pollingInterval,
+}: UseSingleFeedParams = {}): UseSingleFeedValue => {
+  const sdk = useDialectSdk();
+
+  const query: FindDappMessageQuery = useMemo(
+    () => ({
+      dappVerified: !nonVerifiedDapps,
+      take,
+      skip,
+    }),
+    [nonVerifiedDapps, skip, take]
+  );
+
+  const { data, error } = useSWR(
+    SINGLE_FEED_CACHE_KEY_FN(query),
+    () => sdk.wallet.messages.findAllFromDapps(query),
+    { refreshInterval: pollingInterval }
+  );
+
+  const isLoading = data === undefined && !error;
```

Comment:
Why specifically `undefined`? Can we simplify to `!data && !error`?

Reply:
it comes from how swr handle things, data === undefined means data wasn't fetched yet, but it could be null if null is fetched. @zhelezkov could you add clarification comment here maybe?

Reply:
hm, I thought I took this from swr doc(But I have just rechecked and wasn't able to find proof), or somewhere from our existing code base
But technically @Kiryous is right I think
@fsher wdyt?

Reply:
hm, yes, I think technically you're right